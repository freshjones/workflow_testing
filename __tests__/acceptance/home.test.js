var webdriver = require('selenium-webdriver');

var By = webdriver.By, until = webdriver.until;

var driver = new webdriver.Builder().
    usingServer('http://localhost:4444/wd/hub').
    withCapabilities({browserName: 'chrome'}).
    build();

test('should be on correct page', function (done) {
  
  driver.get("http://localhost:8000");

  getTitle()
    .then(assertTitleIsCorrect)
    .then(getPageBody)
    .then(assertPageSnapshot)
    .then(function(){
      driver.quit()
      done();
    })

  function getTitle() {
    return driver.wait(driver.getTitle(), 10000);
  }

  function getPageBody()
  {
    return driver.findElement( By.tagName('body') ).getAttribute("innerHTML");
  }

  function assertPageSnapshot(body) {
    try{ 
      //var body = driver.findElement( By.tagName('body') );
      expect(body).toMatchSnapshot()
    }
    catch(e){ 
      console.log(e);  
    }
    
  }

  function assertTitleIsCorrect(title) {
    try{ 
      expect(title).toBe('Hello World | My App Name'); 
    }
    catch(e){ 
      console.log(e);  
    }
  }

});
