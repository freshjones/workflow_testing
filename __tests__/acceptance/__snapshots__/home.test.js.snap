// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should be on correct page 1`] = `
"
        <div class=\\"saasafari-auth-app\\"><div data-reactroot=\\"\\" data-reactid=\\"1\\" data-react-checksum=\\"-662995952\\"><!-- react-empty: 2 --><nav data-reactid=\\"3\\"><ul data-reactid=\\"4\\"><li data-reactid=\\"5\\"><a class=\\"active\\" style=\\"color:limegreen;\\" href=\\"/\\" data-reactid=\\"6\\">Homepage</a></li><li data-reactid=\\"7\\"><a href=\\"/other\\" data-reactid=\\"8\\">Other</a></li></ul></nav><div data-reactid=\\"9\\"><!-- react-empty: 10 --><h1 data-reactid=\\"11\\">HomePage</h1></div></div></div>
        <script src=\\"http://localhost:7000/dist/js/bundle.js\\"></script>
      
    "
`;
